client/rest-high-level/src/test/java/org/opensearch/client/AbstractResponseTestCase.java

```diff
@@ -116,4 +116,17 @@ protected static <T> void assertMapEquals(ImmutableOpenMap<String, T> expected,
             assertEquals(expected.get(key), actual.get(key));
         }
     }
+
+    protected static <T> void assertMapEquals(final Map<String, T> expected, Map<String, T> actual) {
+        Set<String> expectedKeys = new HashSet<>();
+        Iterator<String> keysIt = expected.keySet().iterator();
+        while (keysIt.hasNext()) {
+            expectedKeys.add(keysIt.next());
+        }
+
+        assertEquals(expectedKeys, actual.keySet());
+        for (String key : expectedKeys) {
+            assertEquals(expected.get(key), actual.get(key));
+        }
```

Comment:
Why is `assertEquals(expected, actual)` not sufficient? Per the contract of java.util.Map, all implementations must implement the logic you have here in their equals methods. We can keep the method here if it avoids a larger refactoring, but I'm not sure that you need to implement any logic here other than call `assertEquals`.

Reply:
It is sufficient! And it's a cleaner refactor. Made the change. Thx @andrross